SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyBuiltin.swift

```diff
@@ -33,7 +33,7 @@ extension BuiltinInst : OnoneSimplifyable {
            .Alignof:
         optimizeTargetTypeConst(context)
       case .DestroyArray:
-        if let elementType = substitutionMap.replacementTypes[0],
+        if let elementType = substitutionMap.replacementTypes[0].canonical.silType,
            elementType.isTrivial(in: parentFunction)
```

Comment:
You need to lower the type here because it might be a function type or metatype, no?

Reply:
Exactly.
I don't use `loweredType`, yet, because I need to add a flag for getting the "maximally abstracted" lowered type. This is what IRGen does for the size/alignment/stride builtins. This makes a difference for metatypes.
`Builtin.size(Bool.Type.self) == 8`. Whereas the "regular" type lowering is a thin metatype which would have a size of 0.